staging/src/k8s.io/legacy-cloud-providers/gce/gce_instances.go

```diff
@@ -458,37 +481,40 @@ func (g *Cloud) getInstancesByNames(names []string) ([]*gceInstance, error) {
 				continue
 			}
 			if found[inst.Name] != nil {
-				klog.Errorf("Instance name %q was duplicated (in zone %q and %q)", inst.Name, zone, found[inst.Name].Zone)
+				klog.Errorf("Instance name %q was duplicated (in zone %q and %q)", inst.Name, zone, found[inst.Name].instance.Zone)
 				continue
 			}
-			found[inst.Name] = &gceInstance{
-				Zone:  zone,
-				Name:  inst.Name,
-				ID:    inst.Id,
-				Disks: inst.Disks,
-				Type:  lastComponent(inst.MachineType),
+			if allNamesSet.Has(inst.Name) {
+				found[inst.Name] = &instanceOrError{
+					instance: &gceInstance{
+						Zone:  zone,
+						Name:  inst.Name,
+						ID:    inst.Id,
+						Disks: inst.Disks,
+						Type:  lastComponent(inst.MachineType),
+					},
+					err: nil,
+				}
+				remaining--
 			}
-			remaining--
 		}
 	}
 
 	if remaining > 0 {
 		var failed []string
-		for k := range found {
-			if found[k] == nil {
-				failed = append(failed, k)
+		for name, instOrError := range found {
+			if instOrError == nil {
+				failed = append(failed, name)
```

Comment:
Failed is now only used to klog.
Given that it's not necessary an error in the current semantic (we're fine with returning error), this should be removed from here.
And that particular log (of failed instances) should be moved to getInstancesByNames function.

Reply:
done